src/main/java/com/fasterxml/jackson/databind/deser/BeanDeserializer.java

```diff
@@ -505,7 +505,12 @@ protected Object _deserializeUsingPropertyBased(final JsonParser p, final Deseri
             // "any property"?
             if (_anySetter != null) {
                 try {
-                    buffer.bufferAnyParameterProperty(_anySetter, propName, _anySetter.deserialize(p, ctxt));
+                    // [databind#4639] Since 2.18.1 AnySetter might not part of the creator, but just some field.
+                    if (_anySetter.isFieldType()) {
+                        buffer.bufferAnyProperty(_anySetter, propName, _anySetter.deserialize(p, ctxt));
+                    } else {
+                        buffer.bufferAnyParameterProperty(_anySetter, propName, _anySetter.deserialize(p, ctxt));
+                    }
```

Comment:
While resolving #562, I did not account for case like #4639 where there is creator, but any-setter is field.